pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -231,9 +267,24 @@ func generateMemoryPatch(vm *kubevirtv1.VirtualMachine, mem *resource.Quantity,
 		if err != nil {
 			return patchOps, err
 		}
+		// already checked on validator: reservedMemory < mem
+		if reservedMemory.Value() >= mem.Value() {
+			return patchOps, fmt.Errorf("reservedMemory cannot be equal or greater than limits.memory %v %v", reservedMemory.Value(), mem.Value())
+		}
+	} else {
+		// if user does not specify the value, then compute it dynamically
+		computeReservedMemory(mem, &reservedMemory)
 	}
 	guestMemory := *mem
 	guestMemory.Sub(reservedMemory)
+	if guestMemory.Value() < 268435456 {
+		// some test cases use a small value to test, but for production such VM make no sense, add a warning here
+		logrus.Warnf("vm %s/%s guest memory is under the suggested minimum requirement(256Mi), original: %v, reserved: %v, final: %v", vm.Namespace, vm.Name, mem.Value(), reservedMemory.Value(), guestMemory.Value())
+	}
+	if guestMemory.Value() < 10485760 {
```

Comment:
For better readability
```suggestion
	if guestMemory.Value() < 10*1024*1024 {
```
or
```suggestion
	if -1 == guestMemory.Cmp(resource.ParseQuantity("10Mi")) {
```